smart_open/tests/test_http.py

```diff
@@ -161,6 +161,14 @@ def test_timeout_attribute(self):
         assert hasattr(reader, 'timeout')
         assert reader.timeout == timeout
 
+    @responses.activate
+    def test_session_attribute(self):
+        session = requests.Session()
+        responses.add_callback(responses.GET, URL, callback=request_callback)
+        reader = smart_open.open(URL, "rb", transport_params={'session': session})
+        assert hasattr(reader, 'session')
+        assert reader.session == session
```

Comment:
can you also call `reader.read()` so the code you touched is covered when a session is passed